src/hooks/use-redirect-when-unlogged.ts

```diff
@@ -0,0 +1,21 @@
+import { useSelector } from "react-redux";
+import { AuthState } from "../features/auth/store/types";
+import { useNavigate } from "react-router-dom";
+import { useEffect } from "react";
+
+export default function useRedirectWhenUnlogged() {
+  const isLogged = useIsLogged();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (!isLogged) {
+      navigate(`/login`);
+    }
```

Comment:
Maybe
```suggestion
    if (!isLogged) navigate(`/login`);
```
Optional, just a preference

Reply:
a mi tb me gusta mas asi